### PR TITLE
feat: enhance frontend

### PR DESCRIPTION
**Pull request by joaodiaslobo**
This pull request includes enhancements to the UI elements across most pages. The modifications consist of an improved team page UI, better player team UI, a more engaging match page with icons to represent player events, a footer, and mobile responsiveness.

All updates have been made in accordance with the figma design file. _Relevant images that showcase the modifications are included below for reference:_

image
image
image
image
image

### CODE REVIEW

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]
Please rebase this PR in order to get main recent changes.

File: lib/cesium_cup_web/live/player_live/show.html.heex [hunk 1]
```diff
@@ -1,45 +1,115 @@
...
+                    <td class="whitespace-nowrap py-4 pr-20 text-sm font-medium text-gray-900 text-right">
+                      <%= if (get_team(match.home_team_id)) do %>
+                        <%= live_redirect to: Routes.team_show_path(@socket, :show, get_team(match.away_team_id)) do %>
+                          <%= get_team(match.away_team_id).name %>
+                        <% end %>
+                      <% end %>
+                    </td>
+                    <td>
+                      <span><%= live_redirect("Show", to: Routes.match_show_path(@socket, :show, match), class: "text-gray-900") %></span>
+                      <span><%= live_patch("Edit", to: Routes.match_index_path(@socket, :edit, match), class: "text-gray-900") %></span>
```
Comment (MarioRodrigues10) [inline lib/cesium_cup_web/live/player_live/show.html.heex:103]:
```suggestion
                      <span><%= live_redirect("Edit", to: Routes.match_index_path(@socket, :edit, match), class: "text-gray-900") %></span>
```

File: lib/cesium_cup_web/live/team_live/show.html.heex [hunk 2]
```diff
@@ -1,94 +1,116 @@
...
+        <.live_component module={CesiumCupWeb.TeamLive.FormComponent} id={@team.id} title={@page_title} action={@live_action} team={@team} return_to={Routes.team_show_path(@socket, :show, @team)} />
+      </.modal>
+    <% end %>
 
-  <div class="flex items-center space-x-10">
-    <div class="h-1/4 w-auto">
-      <img class="w-full" src={Routes.static_path(@socket, "/images/teams/portugal.png")} />
+    <div class="py-4 flex px-4 font-medium text-xs divide-x-4">
+      <p>CeSIUM Cup 2023 > <%= @team.name %></p>
+      <span class="border-transparent ml-auto mr-4 text-indigo-300 hover:text-indigo-400 whitespace-nowrap border-b-4 font-small"><%= live_patch("Edit team", to: Routes.team_show_path(@socket, :edit, @team), class: "button") %></span>
```
Comment (MarioRodrigues10) [inline lib/cesium_cup_web/live/team_live/show.html.heex:11]:
```suggestion
      <span class="border-transparent ml-auto mr-4 text-indigo-300 hover:text-indigo-400 whitespace-nowrap border-b-4 font-small"><%= live_redirect("Edit team", to: Routes.team_show_path(@socket, :edit, @team), class: "button") %></span>
```

File: lib/cesium_cup_web/live/match_live/show.html.heex [hunk 3]
```diff
@@ -1,28 +1,20 @@
...
+<div>
+  <div class="relative pb-4 bg-white mt-16 text-secondary sm:mx-auto rounded-lg sm:px-10">
+    <%= if @live_action in [:edit] do %>
+      <.modal return_to={Routes.match_show_path(@socket, :show, @match)}>
+        <.live_component module={CesiumCupWeb.MatchLive.FormComponent} id={@match.id} title={@page_title} action={@live_action} match={@match} return_to={Routes.match_show_path(@socket, :show, @match)} />
+      </.modal>
+    <% end %>
+    <div class="py-4 px-4 flex font-medium text-xs divide-x-4">
+      <p>CeSIUM Cup 2023 > <%= @match.home_team.name %> vs <%= @match.away_team.name %></p>
+      <span class="border-transparent ml-auto mr-4 text-indigo-300 hover:text-indigo-400 whitespace-nowrap border-b-4 font-small"><%= live_patch("Edit match", to: Routes.match_show_path(@socket, :edit, @match), class: "button") %></span>
```
Comment (MarioRodrigues10) [inline lib/cesium_cup_web/live/match_live/show.html.heex:10]:
```suggestion
      <span class="border-transparent ml-auto mr-4 text-indigo-300 hover:text-indigo-400 whitespace-nowrap border-b-4 font-small"><%= live_redirect("Edit match", to: Routes.match_show_path(@socket, :edit, @match), class: "button") %></span>
```

File: lib/cesium_cup_web/live/group_live/show.html.heex [hunk 4]
```diff
@@ -1,11 +1,16 @@
...
+<div class="pt-10 px-4">
+  <div class="mt-5 bg-[#F9FAFB] rounded-lg w-full px-10 pb-10">
+    <%= if @live_action in [:edit] do %>
+      <.modal return_to={Routes.group_show_path(@socket, :show, @group)}>
+        <.live_component module={CesiumCupWeb.GroupLive.FormComponent} id={@group.id} title={@page_title} action={@live_action} group={@group} return_to={Routes.group_show_path(@socket, :show, @group)} />
+      </.modal>
+    <% end %>
+    <div class="py-4 px-4 flex text-secondary font-medium text-xs divide-x-4">
+      <p>CeSIUM Cup 2023 > Group <%= @group.name %></p>
+      <span class="border-transparent ml-auto mr-4 text-indigo-300 hover:text-indigo-400 whitespace-nowrap border-b-4 font-small"><%= live_patch("Edit group", to: Routes.group_show_path(@socket, :edit, @group), class: "button") %></span>
```
Comment (MarioRodrigues10) [inline lib/cesium_cup_web/live/group_live/show.html.heex:10]:
```suggestion
      <span class="border-transparent ml-auto mr-4 text-indigo-300 hover:text-indigo-400 whitespace-nowrap border-b-4 font-small"><%= live_redirect("Edit group", to: Routes.group_show_path(@socket, :edit, @group), class: "button") %></span>
```

File: lib/cesium_cup_web/live/player_live/show.html.heex [hunk 5]
```diff
@@ -1,45 +1,115 @@
...
+    <% end %>
 
-  <%= if @live_action in [:edit] do %>
-    <.modal return_to={Routes.player_show_path(@socket, :show, @player)}>
-      <.live_component module={CesiumCupWeb.PlayerLive.FormComponent} id={@player.id} title={@page_title} action={@live_action} player={@player} return_to={Routes.player_show_path(@socket, :show, @player)} />
-    </.modal>
-  <% end %>
+    <div class="py-4 px-4 flex font-medium text-xs divide-x-4">
+      <p>CeSIUM Cup 2023 > <%= @player.team.name %> > <%= @player.name %></p>
+      <span class="border-transparent ml-auto mr-4 text-indigo-300 hover:text-indigo-400 whitespace-nowrap border-b-4 font-small"><%= live_patch("Edit player", to: Routes.player_show_path(@socket, :edit, @player), class: "button") %></span>
```
Comment (MarioRodrigues10) [inline lib/cesium_cup_web/live/player_live/show.html.heex:11]:
```suggestion
      <span class="border-transparent ml-auto mr-4 text-indigo-300 hover:text-indigo-400 whitespace-nowrap border-b-4 font-small"><%= live_redirect("Edit player", to: Routes.player_show_path(@socket, :edit, @player), class: "button") %></span>
```

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]

**Review by MarioRodrigues10 — CHANGES_REQUESTED** [review]
Please run ```mix format``` in order to pass CI

**Review by MarioRodrigues10 — DISMISSED** [review]
There's a few changes you'll have to do:
1st - make sure every page design is responsive
2nd - ```/players``` could you enhance this page
3rd - please enhance ```edit``` pages

**Comment by joaodiaslobo** [comment]
Changed player page to match Figma design.

image

File: lib/cesium_cup_web/live/player_live/show.ex [hunk 6]
```diff
@@ -11,10 +12,60 @@ defmodule CesiumCupWeb.PlayerLive.Show do
...
+      )
+    )
+  end
+
+  defp get_age_string(birthday) do
+    today = Date.utc_today()
+    age = Timex.diff(today, birthday, :year)
+    birthday_string = display_date(birthday)
+    "#{age} (#{birthday_string})"
+  end
```
Comment (ruilopesm) [inline lib/cesium_cup_web/live/player_live/show.ex]: This function should be inside an helpers module 💡

Comment (ruilopesm) [reply lib/cesium_cup_web/live/player_live/show.ex]: From what I can see, `ViewUtils` already exist. Put it there 👌🏼

**Review by feliciofilipe — APPROVED** [review]
🙏🙏

**Pull request merged**